src/battle/battle_system.c

```diff
@@ -1303,3 +1331,634 @@ u32 CalcMoneyLoss(Party *party, PlayerProfile *profile) {
+    BufferMoveName(bsys->msgFormat, bufferIndex, param);
+}
+
+static void BattleMessage_BufferItem(BattleSystem *bsys, int bufferIndex, int param) {
+    BufferItemName(bsys->msgFormat, bufferIndex, param);
+}
+
+static void BattleMessage_BufferNumber(BattleSystem *bsys, int bufferIndex, int param) {
+    BufferIntegerAsString(bsys->msgFormat, bufferIndex, param, 5, PRINTING_MODE_LEFT_ALIGN, TRUE);
+}
+
+static void BattleMessage_BufferNumbers(BattleSystem *bsys, int bufferIndex, int param, int a3) {
+    if (a3) {
+        BufferIntegerAsString(bsys->msgFormat, bufferIndex, param, a3, PRINTING_MODE_RIGHT_ALIGN, TRUE);
+    } else {
+        BufferIntegerAsString(bsys->msgFormat, bufferIndex, param, 5, PRINTING_MODE_RIGHT_ALIGN, TRUE);
+    }
+    
+}
+
+static void BattleMessage_BufferType(BattleSystem *bsys, int bufferIndex, int param) {
+    BufferTypeName(bsys->msgFormat, bufferIndex, param);
+}
+
+static void BattleMessage_BufferAbility(BattleSystem *bsys, int bufferIndex, int param) {
+    BufferAbilityName(bsys->msgFormat, bufferIndex, param);
+}
+
+static void BattleMessage_BufferStat(BattleSystem *bsys, int bufferIndex, int param) {
+    BufferStatName(bsys->msgFormat, bufferIndex, param);
+}
+
+static void BattleMessage_BufferStatus(BattleSystem *bsys, int bufferIndex, int param) {
+    BufferStatusName(bsys->msgFormat, bufferIndex, param);
+}
+
+static void BattleMessage_BufferPokemon(BattleSystem *bsys, int bufferIndex, int param) {
+    Pokemon *mon = BattleSystem_GetPartyMon(bsys, param & 0xFF, (param & 0xFF00) >> 8);
+    BufferBoxMonSpeciesName(bsys->msgFormat, bufferIndex, &mon->box);
+}
+
+static void BattleMessage_BufferPoffin(BattleSystem *bsys, int bufferIndex, int param) {
+    //poffins don't exist in HGSS
+}
+
+static void BattleMessage_BufferFlavorPreference(BattleSystem *bsys, int bufferIndex, int param) {
+    BufferFlavorDislikeText(bsys->msgFormat, bufferIndex, param);
+}
+
+static void BattleMessage_BufferTrainerClass(BattleSystem *bsys, int bufferIndex, int param) {
+    Trainer *trainer = BattleSystem_GetTrainer(bsys, param);
+    BufferTrainerClassNameFromDataStruct(bsys->msgFormat, bufferIndex, trainer);
+}
+
+static void BattleMessage_BufferTrainerName(BattleSystem *bsys, int bufferIndex, int param) {
+    Trainer *trainer = BattleSystem_GetTrainer(bsys, param);
+    BufferTrainerNameFromDataStruct(bsys->msgFormat, bufferIndex, trainer);
+}
+
+static void BattleMessage_BufferBoxName(BattleSystem *bsys, int bufferIndex, int param) {
+    BufferPCBoxName(bsys->msgFormat, bufferIndex, bsys->storage, param);
+}
+
+static void BattleMessage_ExpandPlaceholders(BattleSystem *bsys, MsgData *data, BattleMessage *msg) {
+    String *str = NewString_ReadMsgData(data, msg->id);
+    StringExpandPlaceholders(bsys->msgFormat, bsys->msgBuffer, str);
+    String_Delete(str);
+}
+
+static BOOL ov12_0223CF14(struct TextPrinterTemplate *template, u16 glyphId) {
```

Comment:
sound control characters?

Reply:
what?

Reply:
is the func checking for sound control characters?

Reply:
no clue, it's the function fed into the text printers in this file when they're being created

Reply:
also remove struct keyword

Reply:
the struct isn't typedef'd and it's out of scope at this point to fix that for each instance where the TextPrtinterTemplate struct occurs